packages/core/src/scheduler/task.rs

```diff
@@ -48,20 +52,25 @@ impl Scheduler {
             })),
         };
 
+        let mut cx = std::task::Context::from_waker(&task.waker);
+        if task.task.borrow_mut().as_mut().poll(&mut cx).is_ready() {
+            return None;
+        }
+
         entry.insert(task);
 
         self.sender
             .unbounded_send(SchedulerMsg::TaskNotified(task_id))
             .expect("Scheduler should exist");
 
-        task_id
+        Some(task_id)
     }
 
     /// Drop the future with the given TaskId
     ///
     /// This does not abort the task, so you'll want to wrap it in an aborthandle if that's important to you
-    pub fn remove(&self, id: TaskId) {
-        self.tasks.borrow_mut().try_remove(id.0);
+    pub fn remove(&self, id: TaskId) -> Option<LocalTask> {
+        self.tasks.try_borrow_mut().ok()?.try_remove(id.0)
```

Comment:
Forgot to change this back when getting tests to pass, but doesn't effect stress tests